### PR TITLE
gateways now pause movement briefly to prevent chain teleports

### LINKED ISSUE #2717: Going through the away portal doesn't invert your movement direction or cancel your movement
You go up to enter the portal. If you hold it down while travelling through the portal, you end up travelling back. This makes you "flash" as you're travelling so fast. To fix this, travelling through the portal could cancel your movement so you have to release and depress the key again to continue or invert your movement on the other side of the portal so you move downwards while holding up until you release the button.

### PR DESCRIPTION
**Pull request by MrsTonedOne**
Fixes #2717

### CODE REVIEW

**Comment by KorPhaeron** [comment]
@MrStonedOne 

You should make this do the return gate only, or only if the target gate is calibrated.

Otherwise you'll run into scenarios where you spawn in a dangerous area but can't move

**Comment by MrStonedOne** [comment]
it already does the latter idea.

**Comment by KorPhaeron** [comment]
Nevermind me then

File: code/modules/awaymissions/gateway.dm [hunk 1]
```diff
@@ -131,7 +131,7 @@ var/obj/machinery/gateway/centerstation/the_gateway = null
 
 
 //okay, here's the good teleporting stuff
-/obj/machinery/gateway/centerstation/Bumped(atom/movable/M as mob|obj)
+/obj/machinery/gateway/centerstation/Bumped(atom/movable/AM as mob|obj)
```
Comment (RemieRichards) [inline code/modules/awaymissions/gateway.dm]: pls remove "as X" nonsense, it's entirely unneeded.

File: code/modules/awaymissions/gateway.dm [hunk 2]
```diff
@@ -235,20 +239,24 @@ var/obj/machinery/gateway/centerstation/the_gateway = null
 	toggleoff()
 
 
-/obj/machinery/gateway/centeraway/Bumped(atom/movable/M as mob|obj)
+/obj/machinery/gateway/centeraway/Bumped(atom/movable/AM as mob|obj)
```
Comment (RemieRichards) [inline code/modules/awaymissions/gateway.dm]: ditto

**Comment by WJohn** [comment]
Uh, wrong issue you're fixing there in your OP.

**Comment by KazeEspada** [comment]
The code is copypasta and terrible, but that's just the way gateways are coded.

**Comment by MrStonedOne** [comment]
Fixed @WJohn 

I'll get to the code stuff later remie

**Pull request merged**